### PR TITLE
Add git features to Dancer2 CLI

### PR DESCRIPTION
**Pull request by cromedome**
This is the same PR as #1596, but redone to account for the new CLI. This offers the same feature set, but this time done with `CLI::Osprey` and `Path::Tiny`.

Please see the original PR for discussion.

### CODE REVIEW

File: lib/Dancer2/CLI/Gen.pm [hunk 1]
```diff
@@ -81,6 +105,12 @@ Invalid application name. Application names must not contain single colons,
 dots, hyphens or start with a number.
     }) unless is_module_name( $self->application );
 
+    if ( my $remote = $self->remote ) {
+        my $scheme = URI->new( $remote )->scheme // $self->remote; # This feels dirty
+        $scheme eq 'git' || $scheme =~ /^http/ || $scheme =~ /^git@.+:.+\.git$/
+          or $self->osprey_usage( 1, "'$remote' must be a valid URI to git repository");
```
Comment (yanick) [inline lib/Dancer2/CLI/Gen.pm:111]: maybe

```suggestion
        $self->osprey_usage( 1, "'$remote' must be a valid URI to git repository")
          unless $scheme =~ / ^ git \@ .+ : .+ \.git $ | ^ http /x;
```

Comment (cromedome) [reply lib/Dancer2/CLI/Gen.pm]: Used this. Thanks!

File: lib/Dancer2/CLI/Gen.pm [hunk 2]
```diff
@@ -81,6 +105,12 @@ Invalid application name. Application names must not contain single colons,
 dots, hyphens or start with a number.
     }) unless is_module_name( $self->application );
 
+    if ( my $remote = $self->remote ) {
+        my $scheme = URI->new( $remote )->scheme // $self->remote; # This feels dirty
+        $scheme eq 'git' || $scheme =~ /^http/ || $scheme =~ /^git@.+:.+\.git$/
```
[the remark this answers is not in the document]
Comment (cromedome) [reply lib/Dancer2/CLI/Gen.pm]: Incorporated above. Thanks!

**Comment by racke** [comment]
There are still http:// links present even after rebasing this branch on master:

If you need community assistance, the following resources are available:
- Dancer website: **http://perldancer.org**
- Twitter: https://twitter.com/PerlDancer/
- GitHub: https://github.com/PerlDancer/Dancer2/
- Mailing list: **http://lists.perldancer.org/mailman/listinfo/dancer-users**
- IRC: irc.perl.org#dancer

**Comment by racke** [comment]
Also we should consider whether we start with "main" as default branch instead of "master".

**Comment by bigpresh** [comment]
> Also we should consider whether we start with "main" as default branch instead of "master".

Maybe let the user decide if they want to bother with that, rather than forcing that on them?

**Comment by racke** [comment]
> > Also we should consider whether we start with "main" as default branch instead of "master".
> 
> Maybe let the user decide if they want to bother with that, rather than forcing that on them?

Fine for me.

**Comment by cromedome** [comment]
> There are still http:// links present even after rebasing this branch on master:
> 
> If you need community assistance, the following resources are available:
> 
> * Dancer website: **http://perldancer.org**
> * Twitter: https://twitter.com/PerlDancer/
> * GitHub: https://github.com/PerlDancer/Dancer2/
> * Mailing list: **http://lists.perldancer.org/mailman/listinfo/dancer-users**
> * IRC: irc.perl.org#dancer

Fixed. Not sure why those got lost in the rebase.

**Comment by cromedome** [comment]
> > Also we should consider whether we start with "main" as default branch instead of "master".
> 
> Maybe let the user decide if they want to bother with that, rather than forcing that on them?
I agree with that. That being said, should we change our master branch to main? Been wondering if I should just pull the trigger on that.

**Comment by bigpresh** [comment]
> I agree with that. That being said, should we change our master branch to main?

Personally, I really don't see the point.

**Comment by yanick** [comment]
>  should we change our master branch to main? 

That would be nice, yeah. :heart:

**Comment by racke** [comment]
> > should we change our master branch to main?
> 
> That would be nice, yeah. heart

At any rate, this question is not relevant to this ticket.

**Comment by cromedome** [comment]
Anyone see anything I missed on this PR? If not, can I get some approvals, and I will merge and release? Thanks :)

**Review by racke — APPROVED** [review]
Let's not stall it further. Good enough to go.

**Review by yanick — APPROVED** [review]
ship it!

**Comment by cromedome** [comment]
Merged. Thanks everyone!

**Pull request closed without merging**